wrapper/src/test/java/integration/refactored/container/tests/ReadWriteSplittingTests.java

```diff
@@ -417,71 +463,244 @@ public void test_setReadOnlyFalse_allInstancesDown() throws SQLException {
+        if (otherReaderId.equals(instanceId)) {
+          continue;
+        }
+        ProxyHelper.disableConnectivity(instanceId);
+      }
+
+      assertFirstQueryThrows(conn, SqlState.COMMUNICATION_LINK_CHANGED.getState());
+      assertFalse(conn.isClosed());
+      String currentConnectionId = queryInstanceId(conn);
+      assertEquals(otherReaderId, currentConnectionId);
+      assertNotEquals(readerConnectionId, currentConnectionId);
+
+      ProxyHelper.enableAllConnectivity();
 
-        stmt.executeQuery("SELECT " + i);
-        conn.rollback();
-        readerConnectionId = queryInstanceId(conn);
-        assertNotEquals(writerConnectionId, readerConnectionId);
+      conn.setReadOnly(false);
+      currentConnectionId = queryInstanceId(conn);
+      assertEquals(writerConnectionId, currentConnectionId);
+
+      conn.setReadOnly(true);
+      currentConnectionId = queryInstanceId(conn);
+      assertEquals(otherReaderId, currentConnectionId);
+    }
+  }
+
+  @TestTemplate
+  @EnableOnDatabaseEngine({DatabaseEngine.MYSQL, DatabaseEngine.PG})
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  @EnableOnTestFeature({TestEnvironmentFeatures.NETWORK_OUTAGES_ENABLED, TestEnvironmentFeatures.FAILOVER_SUPPORTED})
+  public void test_failoverReaderToWriter_setReadOnlyTrueFalse() throws SQLException {
+    try (final Connection conn =
+             DriverManager.getConnection(getProxiedUrl(), this.defaultProps)) {
+
+      final String writerConnectionId = queryInstanceId(conn);
+
+      conn.setReadOnly(true);
+      final String readerConnectionId = queryInstanceId(conn);
+      assertNotEquals(writerConnectionId, readerConnectionId);
+
+      List<TestInstanceInfo> instances = TestEnvironment.getCurrent().getInfo().getDatabaseInfo().getInstances();
+
+      // Kill all instances except the writer
+      for (TestInstanceInfo instance : instances) {
+        final String instanceId = instance.getInstanceId();
+        if (writerConnectionId.equals(instanceId)) {
+          continue;
+        }
+        ProxyHelper.disableConnectivity(instanceId);
       }
+
+      assertFirstQueryThrows(conn, SqlState.COMMUNICATION_LINK_CHANGED.getState());
+      assertFalse(conn.isClosed());
+      String currentConnectionId = queryInstanceId(conn);
+      assertEquals(writerConnectionId, currentConnectionId);
+
+      ProxyHelper.enableAllConnectivity();
+
+      conn.setReadOnly(true);
+      currentConnectionId = queryInstanceId(conn);
+      assertNotEquals(writerConnectionId, currentConnectionId);
+
+      conn.setReadOnly(false);
+      currentConnectionId = queryInstanceId(conn);
+      assertEquals(writerConnectionId, currentConnectionId);
     }
   }
 
+  // TODO: need to fix
+//   @TestTemplate
+//   @EnableOnDatabaseEngine({DatabaseEngine.MYSQL, DatabaseEngine.PG})
+//   @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+//   public void test_multiHostUrl_topologyOverridesHostList() throws SQLException {
```

Comment:
I believe this test is outdated and we can remove it. Originally it was meant to test that the driver would automatically override a comma-delimited static host list with the dynamic topology fetched from Aurora. However, the current implementation is different and we have already documented that users should not supply a comma-delimited list if they are connecting to an aurora cluster.

Reply:
great, thanks!